### DISCUSSION TITLE
Authorization on the Action trigger button. How to?

### DISCUSSION THREAD

**Discussion by cleargoal** [filamentphp/filament#12611] [Help] [answered]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

I try to check an authorization of custom Action, like this
```
                Action::make('sell')
                    ->url(fn (Vehicle $record): string => route('filament.investor.resources.vehicles.sell', $record))
                    ->visible(auth()->user()->can('update', $record))
                    ->button()->color('success')
```
but actually, I don't understand what to do :(

There is an error **Undefined variable $record** on the line `->visible(auth()->user()->can('update', $record))` but I wondered - why? 
It would seem the variable already exists at the previous line.

Also, I doubt the route name because 'filament.investor.resources.vehicles.sell' I set by analogy as 'filament.investor.resources.vehicles.edit', and I am not sure that it may work.

Clear these moments for me, please.

**Comment by leandrocfe** [comment, marked as answer]
You should also inject it
```php
 ->visible(fn(Vehicle $record) => auth()->user()->can('update', $record))
```

**Reply by cleargoal** [reply]
I found another way, accidentally. It's simple and good work:
```
                Action::make('sell')
                    ->authorize('sell')
```
Also, created the VehiclePolicy custom method 'sell'.
